internal/taskhandlers/create_task.go

```diff
@@ -0,0 +1,70 @@
+package taskhandlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/L0Qqi/To-Do_list/internal/app"
+	"github.com/L0Qqi/To-Do_list/internal/domain/models"
+	"github.com/L0Qqi/To-Do_list/internal/domain/services/nextDate"
+)
+
+// Создание задачи
+func PostTaskHandler(app *app.App) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		//func (app *App) PostTaskHandler(w http.ResponseWriter, r *http.Request) {
+		var task models.Task
+
+		//Декодируем запрос
+		if err := json.NewDecoder(r.Body).Decode(&task); err != nil {
+			http.Error(w, `{"error": "Ошибка в декодировании JSON"}`, http.StatusBadRequest)
+			return
+		}
+		//Если название пустое, вернем ошибку
+		if task.Title == "" {
+			http.Error(w, `{"error": "Поле title не может быть пустым"}`, http.StatusBadRequest)
+			return
+		}
+
+		//Парсим дату, если она есть, иначе дата будет текущей
+		now := time.Now()
+		if task.Date != "" {
+			if _, err := time.Parse("20060102", task.Date); err != nil {
+				http.Error(w, `{"error": "Неверный формат поля date"}`, http.StatusBadRequest)
+				return
+			}
+		} else {
+			task.Date = now.Format("20060102")
+		}
+
+		//Вычисляем следующую дату
+		nextDate, err := nextDate.NextDateAdd(now, task.Date, task.Repeat)
```

Comment:
Лучше в функциях не писать имя пакета, чтобы лучше смотрелось
```suggestion
		nextDate, err := nextDate.Add(now, task.Date, task.Repeat)
```